### PR TITLE
Api

### PR DESCRIPTION
**Pull request by wacoo**
- Add api end-point to posts
- Add api end-point to comments
- Add api end-point to create comments
- Fix linter errors

### CODE REVIEW

File: app/controllers/api/v1/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Api::V1::PostsController < ApplicationController
+  skip_before_action :authenticate_user!
```
Comment (mckent05) [inline app/controllers/api/v1/posts_controller.rb]: - You have done well so far, but please do not skip authentication for your controller actions, as we only want registered users to have access to the app. Please fix this 🙏 
**OPTIONAL** - To help with authentication for your rails api, I suggest you use the devise_jwt for authentication, you can follow this [guide](https://enlear.academy/how-to-create-a-rails-6-api-with-devise-jwt-46fa35085e85) on how to install devise_jwt in your project, please fix this 😄

File: app/controllers/api/v1/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+class Api::V1::CommentsController < ApplicationController
+  skip_before_action :authenticate_user!
+  protect_from_forgery unless: -> { request.format.json? }
+  def index
+    @post = Post.find(params[:post_id])
+    @comments = Comment.where(post: @post)
+    render json: @comments
```
Comment (mckent05) [inline app/controllers/api/v1/comments_controller.rb]: - You have done well so far, but according to the project requirement, you are to create an API endpoint to list all comments for a user's post, from your routes for this action, `/api/v1/users/:user_id/posts/:post_id/comments`, we need to find the user using the user_id, then find the specific post belonging to that user. After finding the post we can get the comments belonging to that post. Please fix this.
```ruby
     @post = Post.find_by(user_id:params[:user_id], post_id:params[:post_id])
     @comments = @posts.comments  
```

File: app/controllers/api/v1/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    @comments = Comment.where(post: @post)
+    render json: Comment.find(params[:id])
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new(comment_params)
+    @comment.user = @user
+    @comment.post = @post
```
Comment (mckent05) [inline app/controllers/api/v1/comments_controller.rb]: - You have done well so far, but according to the project requirements, you are to create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user who makes it; it should not be set by passing a parameter. From what you have done you are setting the user creating the comments. Please fix this 🙏 
```ruby
     @post = Post.find_by(user_id:params[:user_id], post_id:params[:post_id])
     new_comment = current_user.comments.new(comment_params)
     new_comment.post = @post
```

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @wacoo, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job setting your your API endpoints 👍 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TadesseDev — APPROVED** [review]
### Hi @wacoo 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
AmazingCreepyFrogmouth-max-1mb




## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**